### ISSUE TITLE
Add serde support

### ISSUE THREAD

**Issue by alberdingk-thijm** [thejpster/rust-measurements#21]
I'd like to be able to serialize and deserialize measurement data without writing serializing/deserializing code for converting back and forth myself.

serde.rs [gives some info](https://serde.rs/feature-flags.html) on how to add serde support as an optional crate feature using feature flags. Essentially, we just need to add a few lines of

```rust
#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
struct ...
```

This doesn't look particularly time-consuming, so I'd be willing to make the changes myself in a few days and make a PR.

**Comment by thejpster** [comment]
Yeah, if it's opt-in then that's great.

**Comment by thejpster** [comment]
Merged #22

**Comment by thejpster** [comment]
See https://crates.io/crates/measurements v0.10.3.
